pkg/daemon/ceph/osd/volume.go

```diff
@@ -541,6 +541,12 @@ func (a *OsdAgent) useRawMode(context *clusterd.Context, pvcBacked bool) (bool,
 		useRawMode = false
 	}
 
+	// TODO: when we re-enable raw mode, we cannot merely remove the below lines. we must handle
+	// device-specific configs which this function doesn't take into account.
+	// see: https://github.com/rook/rook/pull/8316#issuecomment-880816972
+	logger.Debugf("won't use raw mode to avoid possibility of phantom atari partitions per Rook issue #7940")
+	useRawMode = false
+
```

Comment:
This is the effective change for this PR.

The rest of the changes remove unnecessary/non-useful code, and there are documentation updates.